### ISSUE TITLE
Remove 'swagger' branding from externally facing docs

### ISSUE THREAD

**Issue by mario-guerra** [Azure/typespec-azure#294]
This issue concerns the terminology used in our TypeSpec documentation. We need to replace all instances of the term "swagger" with "OpenAPI specification".

The term "swagger" is a registered trademark of SmartBear, and its usage in our external documentation could potentially lead to trademark issues. Therefore, it's essential that we refer to these as "OpenAPI specifications" in all our external-facing documents.

This change is particularly urgent for the "TypeSpec Azure" section, but it should also be implemented across all other pages in our documentation.

**Comment by timotheeguerin** [comment]
what applies to this? any reference to swagger or only when talking about a swagger spec. For example the playground can show the Swagger UI which is named that way(and the dropdown shows it that way)

**Comment by mario-guerra** [comment]
Anything externally facing needs to be scrubbed since Swagger is a trademark owned by someone else.

**Comment by timotheeguerin** [comment]
yeah but for the case of swagger ui what do we call it, it is called swagger ui, the package is called `swagger-ui`, the Ui will show up swagger ui.

**Comment by timotheeguerin** [comment]
Yeah its the official swagger-ui package https://www.npmjs.com/package/swagger-ui
<img width="2303" alt="image" src="https://github.com/Azure/typespec-azure/assets/1031227/b226766b-48ec-4704-b453-aa1e288c68d3">
